### PR TITLE
Allwrp

### PR DESCRIPTION
**Pull request by valpesendorfer**
Address #143  and #145 

Happy to hear your thoughts @interob

### CODE REVIEW

File: modape/modis/window.py [hunk 1]
```diff
@@ -139,18 +139,23 @@ def generate_mosaics(self,
...
 
         elif target_srs == "EPSG:4326":
 
-            if not attrs["globalproduct"]:
-                output_res = attrs["resolution"] / 112000
-            else:
-                output_res = attrs["resolution"]
+            try:
+                if not attrs["globalproduct"]:
+                    output_res = attrs["resolution"] / 112000
```
Comment (valpesendorfer) [inline modape/modis/window.py:151]: We default back now to `[None, None]` if none of the conditions apply or if we can't read the info from file.

File: modape/modis/window.py [hunk 2]
```diff
@@ -70,7 +70,7 @@ def __init__(self,
     def generate_mosaics(self,
                          dataset,
                          targetdir,
-                         target_srs: str,
+                         target_srs: str = None,
```
Comment (valpesendorfer) [inline modape/modis/window.py:73]: If target_srs is `None`, the src projection will be used

File: modape/scripts/modis_window.py [hunk 3]
```diff
@@ -159,6 +160,9 @@ def cli(src: str,
     if round_int is not None:
         round_int = round_int * -1
 
+    if target_srs.lower() == "none":
```
Comment (valpesendorfer) [inline modape/scripts/modis_window.py:163]: The user can basically specify `--target-srs None` or `none` or `NoNe` ... and it'll be evaluated to `None`.

This way we can keep the convenient default of 4326, while enabling using the existing srs without specifying the actual reference

**Review by interob — APPROVED** [review]
I can see your preferred way of coding is not to initialise a variable in the common (inherited) scope to all references / usages. I do prefer it that way, that's why I initialised output_res to [None, None] on line 142 -- maybe this is an habit I developed in languages like C/C++/C#...

Anyway, I am Approving trusting that all code paths lead through an assignment to (and implicit declaration of) the output_res variable ;-)

**Comment by valpesendorfer** [comment]
> I can see your preferred way of coding is not to initialise a variable in the common (inherited) scope to all references / usages. I do prefer it that way, that's why I initialised output_res to [None, None] on line 142 -- maybe this is an habit I developed in languages like C/C++/C#...
> 
> Anyway, I am Approving trusting that all code paths lead through an assignment to (and implicit declaration of) the output_res variable ;-)

I was looking to add an `else` condition and all cases not covered above should result in `[None, None]`. But if you prefer, we can also initialize above and then just `pass` in else.

**Comment by interob** [comment]
> I was looking to add an else condition and all cases not covered above should result in [None, None]. But if you prefer, we can also initialize above and then just pass in else.

https://github.com/WFP-VAM/modape/blob/ed94650259280ee0f29110ce64ba10a82aa43292/modape/modis/window.py#L142-L158

Hm, why are the conditions `"xRes" in kwargs and "yRes" in kwargs` and `target_srs == "EPSG:4326"` related anyway?

**Comment by valpesendorfer** [comment]
> > I was looking to add an else condition and all cases not covered above should result in [None, None]. But if you prefer, we can also initialize above and then just pass in else.
> 
> https://github.com/WFP-VAM/modape/blob/ed94650259280ee0f29110ce64ba10a82aa43292/modape/modis/window.py#L142-L158
> 
> Hm, why are the conditions `"xRes" in kwargs and "yRes" in kwargs` and `target_srs == "EPSG:4326"` related anyway?

If it's 4326, we can "calculate" the degrees by dividing through 112000. Never checked how true this holds for other GCS.

**Comment by interob** [comment]
>If it's 4326, we can "calculate" the degrees by dividing through 112000. Never checked how true this holds for other GCS.

Yes. But, a user could also want to divert from the "native" resolution. Sometimes for ease of further processing / analysis it's useful to subsample, oversample, resample/realign.

**Comment by valpesendorfer** [comment]
> > If it's 4326, we can "calculate" the degrees by dividing through 112000. Never checked how true this holds for other GCS.
> 
> Yes. But, a user could also want to divert from the "native" resolution. Sometimes for ease of further processing / analysis it's useful to subsample, oversample, resample/realign.

The user can independently specify both resolution and projection.

It's just if no resolution is specified, and if EPSG is 4326, the resolution is calculated in this way (rather than through gdal).

**Comment by interob** [comment]
> So if you want to divert, just specify a resolution.

Ah, yes. Sorry, I see now. That would indeed work just fine.

**Pull request merged**